iceoryx_utils/test/moduletests/test_relative_pointer_data.cpp

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2019 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_utils/internal/relocatable_pointer/relative_pointer_data.hpp"
+
+#include "test.hpp"
+
+namespace
+{
+using namespace ::testing;
+using namespace iox::rp;
+
+TEST(RealtivePointerData_test, DefaultConstructedResultsInNullptrIdAndOffset)
+{
+    RelativePointerData sut;
+
+    EXPECT_EQ(sut.id(), RelativePointerData::NULL_POINTER_ID);
+    EXPECT_EQ(sut.offset(), RelativePointerData::NULL_POINTER_OFFSET);
+}
+
+TEST(RealtivePointerData_test, DefaultConstructedResultsInLogicallyNullptr)
+{
+    RelativePointerData sut;
+
+    EXPECT_TRUE(sut.isLogicalNullptr());
+}
+
+TEST(RealtivePointerData_test, ResetOnDefaultConstructedResultsInLogicallyNullptr)
+{
+    RelativePointerData sut;
+
+    sut.reset();
+
+    EXPECT_TRUE(sut.isLogicalNullptr());
+}
+
+TEST(RealtivePointerData_test, ConstructedWithNonZeroIdAndOffsetResultsInSameIdAndOffset)
+{
+    constexpr uint16_t ID{13U};
+    constexpr uint64_t OFFSET{42U};
+
+    RelativePointerData sut{ID, OFFSET};
+
+    EXPECT_EQ(sut.id(), ID);
+    EXPECT_EQ(sut.offset(), OFFSET);
+}
+
+TEST(RealtivePointerData_test, ConstructedWithZeroIdAndOffsetResultsNotInLogicallyNullptr)
+{
+    constexpr uint16_t ID{0U};
+    constexpr uint64_t OFFSET{0U};
+
+    RelativePointerData sut{ID, OFFSET};
+
+    EXPECT_FALSE(sut.isLogicalNullptr());
+}
+
+TEST(RealtivePointerData_test, ConstructedWithNonZeroIdAndOffsetResultsNotInLogicallyNullptr)
+{
+    constexpr uint16_t ID{13U};
+    constexpr uint64_t OFFSET{42U};
+
+    RelativePointerData sut{ID, OFFSET};
+
+    EXPECT_FALSE(sut.isLogicalNullptr());
+}
+
+TEST(RealtivePointerData_test, ConstructedWithMaxIdAndOffsetResultsNotInLogicallyNullptr)
+{
+    constexpr uint16_t ID{RelativePointerData::MAX_VALID_ID};
+    constexpr uint64_t OFFSET{RelativePointerData::MAX_VALID_OFFSET};
```

Comment:
Could you please add also a test for the limits.
```
    constexpr uint16_t ID{RelativePointerData::MAX_VALID_ID - 1U};
    constexpr uint64_t OFFSET{RelativePointerData::MAX_VALID_OFFSET - 1U};
```
This should result in a non logical nullptr.

Reply:
I can do it, but I do not see a gain in having this tests. What is your intention with those tests?

Reply:
To see that it is really working. Lets assume the limits are wrong it and the MAX_VALID_OFFSET is a 54 bit number then it would override some parts of the id and the whole system could crash.

Reply:
already tested with another test